src/main/java/com/Others/FastPower.java

```diff
@@ -0,0 +1,37 @@
+package src.main.java.com.Others;
+
+import java.math.BigInteger;
+
+/**
+ * We may calculate power with loops, but what if the index is too large ?
+ * FastPower aims to calculate quickly in this circumstances with time complexity O(log k),
+ * where k is the index.
+ *
+ * @author DDullahan
+ */
+public class FastPower {
+    public static BigInteger calculate(BigInteger n, BigInteger k, BigInteger mod) {
+        BigInteger ans = BigInteger.ONE;
+        while (!k.equals(BigInteger.ZERO)) {
+            int odd = k.mod(new BigInteger("2")).compareTo(BigInteger.ZERO);
+            if(odd == 1){
+                ans = ans.multiply(n).mod(mod);
+            }
+            k = k.divide(new BigInteger("2"));
```

Comment:
Instead of using `new BigInteger("2")` in a loop, we could declare it as a constant outside:

```java
public static final BigInteger TWO = new BigInteger("2");
```